src/coreclr/jit/morph.cpp

```diff
@@ -17526,10 +17526,15 @@ bool Compiler::fgCheckStmtAfterTailCall()
 //
 bool Compiler::fgCanTailCallViaJitHelper()
 {
-#if !defined(TARGET_X86) || defined(UNIX_X86_ABI) || defined(FEATURE_READYTORUN)
```

Comment:
did it basically turn helper-based tailcalls off for jit?

Reply:
If so I guess we need more (if we have any) microbenchmarks written in F#

Reply:
(I thought I responded but can't see my comment..?)

> did it basically turn helper-based tailcalls off for jit?

No, it just made x86 use the same mechanism we use on all other platforms.

> If so I guess we need more (if we have any) microbenchmarks written in F#

That would be great, and yes, this would probably have shown up as a noticeable perf regression.